planet/api/http.py

```diff
@@ -13,135 +13,170 @@
 # the License.
 
 """Functionality to perform HTTP requests"""
-
-# import http.client as http_client
+import asyncio
 import logging
-import os
-import re
-import time
 
-from requests import Session
-from requests.compat import urlparse
+import httpx
 
 from . import exceptions, models
 from . __version__ import __version__
 
-
-LOGGER = logging.getLogger(__name__)
-
-DEFAULT_CHUNK_SIZE = 32 * 1024
-
-USE_STRICT_SSL = not (os.getenv('DISABLE_STRICT_SSL', '').lower() == 'true')
-
+RETRY_COUNT = 5
 RETRY_WAIT_TIME = 1  # seconds
 
+LOGGER = logging.getLogger(__name__)
 
-# def setup_logging():
-#     log_level = LOGGER.getEffectiveLevel()
-#     urllib3_logger = logging.getLogger(
-#         'requests.packages.urllib3')
-#     urllib3_logger.setLevel(log_level)
-#
-#     # if debug level set then its nice to see the headers of the request
-#     if log_level == logging.DEBUG:
-#         http_client.HTTPConnection.set_debuglevel(1)
-#     else:
-#         http_client.HTTPConnection.set_debuglevel(0)
-#
 
-def _log_request(req):
-    LOGGER.info('%s %s %s %s', req.method, req.url, req.params, req.data)
+class APlanetSession():
```

Comment:
What are your thoughts on renaming this to something like `AsyncPlanetSession` to be more explicit and descriptive?

Reply:
Yeah, this is a good question. One I have as well. All the async stuff I have prepended with `A`. A lot of libraries either use `A` or `Async`. I am ambivalent - one is shorter, the other is more explicit. If it would help you or anyone else orient to what's going on, I am fully on-board with moving to `Async`.

Reply:
@ericrdunham do you have any thoughts on this from your experience orienting to this library?

Reply:
**TL;DR: let the namespace do the thing, and not the name of the class/function/whatever.**

My thoughts would be to remove it entirely and have the documentation push the fact that it's async over and over! The named classes in Python's `asyncio` module don't have any sort of prefix, and I can imagine myself as a user going "okay, I get it" after the umpteenth time of typing it out. :)

I was thinking yesterday about having something like `import planet.sync as planet` for people that didn't want to use the async stuff, and basically everything in there would be a generated/decorated version of everything in the base `planet` module, but it would all be decorated/returned with `asyncio.run`, which would allow a user to do stuff like:

```python
import asyncio

import planet

order = ...
oid = asyncio.run(planet.create_order(planet.get_auth(), order))
print(oid)
```

... as well as ...

```python
import planet.sync as planet

order = ...
oid = planet.create_order(planet.get_auth(), order)
print(oid)
```

I hope that helps!

Reply:
@ericrdunham this pattern looks really good to me vs APlanetSession or AsyncPlanetSession (or ASession/AsyncSession). @strixcuriosus how does it look to you?

One note, we don't actually have a function `planet.create_order()` and I don't see us having one, at least for async operations. This is because communication via the api must be done within the context of a session (I realize it may be beneficial to document why this is the case in our own documentation. For now, I'll reference [httpx](https://www.python-httpx.org/advanced/)). For sync it is simply inefficient to not group all communication within a session (due to having to build/destroy the session each time). For async all concurrent communication must be grouped within a session because there can only be one active session. I'm going to add this information to the docs.

I am also leaning toward keeping specific client functions namespaced and not elevating them. So, a usage I'd propose is:

```python
import planet

order = ...
with planet.Session as session:
    cl = planet.OrdersClient(session)  # there will be other clients, DataClient, TaskingClient, etc..
    oid = cl.create_order(planet.get_auth(), order)
```

Reply:
adding additional information regarding usage of the session will be covered as a part of #242

Reply:
@jreiberkyle I apologize for any confusion! I was mentioning `planet.create_order` as a example convenience thing to maybe make things easier for some people in the future, but not something that at all was expected soon-ish :). I was noodling on the two separate use cases of someone who wants to play around with the library and see what's what as opposed to someone who already knows that they want to use it and use it a lot. :)

I was thinking of the different between something like `requests.get` and the [more advanced `requests.Session` usage](https://requests.readthedocs.io/en/master/user/advanced/#session-objects). The thought for me was that `planet.create_order` could delegate to a new session and a new client for the easy use case, but if someone wants to make ten million orders, then the persistent session and persistent client (i.e. not using `with`) could be valuable and the existing request/session pattern seems like it's already pretty well-established to me, so piggybacking on that would hopefully be less of a learning curve for folks.

With some code:
```python
import planet

# an example for tutorials, notebooks, etc. maybe?
order = ...
oid = planet.sync.create_order(planet.get_auth(), order)  # or the equivalent with e.g. asynio.run(...)
print(oid)

# an example for batch stuff maybe?
orders = read_ten_million_orders_from_somewhere()
session = planet.Session(planet.get_auth())
client = planet.OrdersClient(session)
oids = await asyncio.gather(map(client.create_order, orders))
print(oids)
```

Reply:
Yeah these convenience functions are nice, I agree. And would be a nice enhancement and is captured in #255. Using a Session as a context manager is the recommended approach with httpx (the http library we build upon) and I feel comfortable having this be the recommended approach with our library. The context manager takes care of cleanup. Without it, the user has to remember to do cleanup on their own. That being said, using the session as a persistent session is actually possible as things are implemented. I will make a note to add documentation on this usage (captured in #242)